tests/pact/fqns.repl

```diff
@@ -77,3 +77,165 @@
+                \within defcap invoke - invoking with an qualified reference \
+                \(no namespace)." (test-mod-redeploy-ref.f))
+
+(expect-failure "Execution expected to fail due to assertion failure within \
+                \defcap invoke - invoking with an fully qualified reference."
+                (free.test-mod-redeploy-ref.f))
+(commit-tx)
+
+; The following module redeployment changed the capability `test` to pass.
+(begin-tx)
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+  (defcap test ()
+    true)
+
+  (defun f ()
+    (with-capability (free.test-mod-redeploy-ref.test)
+      1))
+
+  (defun f1 ()
+    (with-capability (test-mod-redeploy-ref.test)
+      1))
+
+  )
+; Before pact-4.8, the capability update (passing `test`) was ignored as the
+; full-qualified reference referenced the previously deployed version of the module.
+(expect-failure "Execution expected to fail because `DisablePact48` will reference \
+                \the prev. deployed version (fully qualified capability) \
+                \- invoking with an unqualified reference." (f))
+
+(expect-failure "Execution expected to fail because `DisablePact48` will reference \
+                \the prev. deployed version (fully qualified capability) \
+                \- invoking with an qualified reference (no namespace)."
+                (test-mod-redeploy-ref.f))
+
+(expect-failure "Execution expected to fail because `DisablePact48` will reference \
+                \the prev. deployed version (fully qualified capability) \
+                \- invoking with a fully qualified reference."
+                (free.test-mod-redeploy-ref.f))
+
+; We repeat almost the same tests but without prefixing the namespace.
+(expect-failure "Execution expected to fail because `DisablePact48` will reference \
+                \the prev. deployed version (qualified capability without namespace) \
+                \- invoking with an unqualified reference."
+                (f1))
+
+(expect-failure "Execution expected to fail because `DisablePact48` will reference \
+                \the prev. deployed version (qualified capability without namespace) \
+                \- invoking with an qualified reference (no namespace)."
+                (test-mod-redeploy-ref.f1))
+
+(expect-failure "Execution expected to fail because `DisablePact48` will reference \
+                \the prev. deployed version (qualified capability without namespace) \
+                \- invoking with a fully qualified reference."
+                (free.test-mod-redeploy-ref.f1))
+(commit-tx)
+
+
+;; Check Pact48 behaviour
+
+; The pact-4.8 update uses the new module reference.
+(begin-tx)
+(env-exec-config []) ; reset
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+; These tests show that failing capabilities work as expected.
```

Comment:
I'm not sure this is doing what you think. **You're still doing an upgrade/redeploy** -- if you really want to demonstrate the difference you would use a new module. I suggest however that this is unnecessary, I think you can delete this whole section, since the success of the next section shows you don't need a "fresh" module to demonstrate the benefit.